src/lib.rs

```diff
@@ -2722,3 +2706,79 @@ fn command_add_output_file(
         cmd.arg("-o").arg(&dst);
     }
 }
+
+static NEW_STANDALONE_ANDROID_COMPILERS: [&str; 4] = [
+    "aarch64-linux-android21-clang",
+    "armv7a-linux-androideabi16-clang",
+    "i686-linux-android16-clang",
+    "x86_64-linux-android21-clang",
+];
+
+fn defines_for_android_compiler(clang_path: &Path) -> Vec<&'static str> {
+    match NEW_STANDALONE_ANDROID_COMPILERS.iter().find(|x| {
+        let x: &OsStr = x.as_ref();
+        x == clang_path.as_os_str()
+    }) {
+        Some(new_clang) => {
+            let mut ret = Vec::with_capacity(2);
+            ret.push("-DANDROID");
```

Comment:
I'd personally sort of prefer to follow whatever the official compilers are doing, if they've stopped passing this I'm not sure if this should explicitly start passing them by default?

Reply:
> if they've stopped passing this I'm not sure if this should explicitly start passing them by default?

1. There are two official way to build code with NDK (as I know): include file for GNU make Android.mk
2. cmake

And both of them is set `-DANDROID`. I suppose this is for compatibility. There are predefined variable `__ANDROID__` that embedded into compiler, but I suppose there are a lot of code that do: `#ifdef ANDROID` instead of `#ifdef __ANDROID__`.

And I am sure that adding dependency to cmake-rs crate is good idea, or generate make file, search `make` is not good idea.

I also thought that __ANDROID_API__ is not defined by default, because of sqlite3.c compilation failed (mmap64 not defined), but looks like problem in cc-rs, not with not seted __ANDROID_API__. The line:
```
                cmd.args.push(format!("--target={}", target).into());
```
should be disabled in cc-rs for new android toolchain. I am comeback with fix.

Reply:
I've updated code. And `__ANDROID_API__` is not used anymore. Though I leave `-DANDROID` for compability with code around, because of ndk-build/NDK's cmake toolchain files define it.